cub/device/dispatch/dispatch_merge_sort.cuh

```diff
@@ -555,21 +559,31 @@ struct DispatchMergeSort : SelectedPolicy
         break;
       }
 
-      int tile_size = MergePolicyT::ITEMS_PER_TILE;
-      OffsetT num_tiles = cub::DivideAndRoundUp(num_items, tile_size);
+      const auto tile_size = MergePolicyT::ITEMS_PER_TILE;
+      const auto num_tiles = CUB_NS_QUALIFIER::DivideAndRoundUp(num_items,
+                                                                tile_size);
 
-      std::size_t block_sort_shmem_size = BlockSortAgentT::SHARED_MEMORY_SIZE;
-      std::size_t merge_shmem_size = MergeAgentT::SHARED_MEMORY_SIZE;
+      const auto block_sort_shmem_size =
+        static_cast<std::size_t>(BlockSortAgentT::SHARED_MEMORY_SIZE);
 
-      std::size_t merge_partitions_size         = (1 + num_tiles) * sizeof(OffsetT);
-      std::size_t temporary_keys_storage_size   = num_items * sizeof(KeyT);
-      std::size_t temporary_values_storage_size = num_items * sizeof(ValueT) * !KEYS_ONLY;
-      std::size_t virtual_shared_memory_size =
-        vshmem_size(max_shmem,
-                    (cub::max)(block_sort_shmem_size, merge_shmem_size),
-                    num_tiles);
+      const auto merge_shmem_size =
+        static_cast<std::size_t>(MergeAgentT::SHARED_MEMORY_SIZE);
 
-      void*  allocations[4]      = {nullptr, nullptr, nullptr, nullptr};
+      const auto merge_partitions_size =
+        static_cast<std::size_t>(1 + num_tiles) * sizeof(OffsetT);
+
+      const auto temporary_keys_storage_size =
+        static_cast<std::size_t>(num_items * sizeof(KeyT));
+
+      const auto temporary_values_storage_size =
+        static_cast<std::size_t>(num_items * sizeof(ValueT)) * !KEYS_ONLY;
+
+      const auto virtual_shared_memory_size = vshmem_size(
+        static_cast<std::size_t>(max_shmem),
+        (CUB_NS_QUALIFIER::max)(block_sort_shmem_size, merge_shmem_size),
```

Comment:
Just use `cub::` here -- the macro is only needed when used outside of the cub namespace. The other usages of `CUB_NS_MACRO` here can also just be `cub::`.

(The Thrust macro below is correct and should remain)